canonical-web-and-design/ubuntu.com
# Pagination and basic search

## Done

- Pagination and sorting
- More data to display multiple pages
- Basic search (description and priority)

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/security/cve
- Query pagination `http://0.0.0.0:8001/security/cve?limit={page-size}` e.g. http://0.0.0.0:8001/security/cve?limit=1, should only show one element.
- Query pagination offset `http://0.0.0.0:8001/security/cve?offset={offset}` e.g. http://0.0.0.0:8001/security/cve?limit=5&offset=5, should show the second page with only 3 elements.
- The previous two values will assume `offset=0` and `limit=20` by default if no value passed (first page from the beginning). Therefore, those 2 parameters are required if you want to navigate to the next page (or if you don't see any CVEs because of lack of data). 
- Sort by date `http://0.0.0.0:8001/security/cve?order_by=oldest`, will sort `http://0.0.0.0:8001/security/cve?order=newest` by default. Accepted values `oldest` and `newest`.
- Search contents of description `http://0.0.0.0:8001/security/cve?search_desc={description}` e.g. http://0.0.0.0:8001/security/cve?search_desc={description}
- Multiple searches example http://0.0.0.0:8001/security/cve?search_desc=gitlab&priority=low
- New data structure for main page has been reverted. Planning to remodel the data for packages and release_status.
- Package search not included.
